inventory/models.py

```diff
@@ -0,0 +1,158 @@
+
+        content_type = ContentType.objects.get_for_model(self.model)
+        ownerships = Ownership.objects.filter(content_type=content_type, is_active=True)
+        if member:
+            ownerships = ownerships.filter(member=member)
+        if group:
+            ownerships = ownerships.filter(group=group)
+        return self.get_queryset().filter(ownerships__in=ownerships).distinct()
+
+
+# File path to upload achievement images to
+def get_item_image_upload_path(instance, filename):
+    # Obtain extension
+    # NB: A file can be renamed to have ANY extension
+    _, extension = os.path.splitext(filename)
+
+    # file will be uploaded to MEDIA_ROOT / images/item/<item_type>/<id>.<file_extension>
+    return 'images/item/{type_str}/{item_id}{extension}'.format(
+        type_str=slugify(instance.__class__.__name__),
+        item_id=instance.id,
+        extension=extension,
+    )
+
+
+class Item(models.Model):
+    """ Item in the inventory system. Abstract root class.
+
+    On permissions:
+    There are several unique default permissions used for Items
+    add_group_ownership_for_<item>: Allows users to add new ownerships to groups they are part of
+    add_member_ownership_for_<item>: Allows users to add new ownerships to any active member
+    maintain_ownerships_for_<item>: Adds additional control rights
+
+    Furthermore the default permissions edit_<item> and delete_<item> are also used in the front-end
+
+    This grants access to assigning items to groups or members respectively
+
+    """
+    name = models.CharField(max_length=128)
+    description = models.TextField(max_length=512, blank=True, null=True)
+    image = models.ImageField(upload_to=get_item_image_upload_path, blank=True, null=True)
+
+    ownerships = GenericRelation('Ownership')
+    # An achievement can also apply to roleplay items
+    achievements = GenericRelation('achievements.AchievementItemLink')
+
+    objects = ItemManager()
+
+    class Meta:
+        abstract = True
+        ordering = ("name",)
+
+        # Set the default permissions. Each item has a couple of addiotional default permissions
+        default_permissions = ('add', 'change', 'delete', 'view',
+                               'add_group_ownership_for',
+                               'add_member_ownership_for',
+                               'maintain_ownerships_for')
+
+    def currently_in_possession(self):
+        """ Returns all ownership items that are currently at the Knights """
+        return self.ownerships.filter(is_active=True)
+
+    def is_owned_by_association(self):
+        """ Returns boolean stating whether this item is owned by the association """
+        return self.ownerships.filter(is_active=True).filter(group__isnull=False).exists()
+
+    def __str__(self):
+        return f'{self.__class__.__name__}: {self.name}'
+
+
+class BoardGame(Item):
+    """ Defines boardgames """
+    bgg_id = models.IntegerField(blank=True, null=True)
```

Comment:
Unfortunately I didn't get it working as javascript doesn't handle xml that well from what I tried, but my original plan was to use this to retrieve information about the boardgame through the Boardgamegeek API. Thus we could also display things like complexity, number of players and stuff like that.
I personally don't see a use to generalize url for any item as the desired url functionality differs. To what should a roleplaying book for instance have an url to? A place where you can buy it? I had a look in the RP inventory system and urls aren't stored there except for price related stuff.

Reply:
The way I vision such a URL component is mainly as a "more information" button. For roleplay books this could be a general info page or even a wiki (what is it about? what type of system is it? how many dice can I roll at once? etc.), as opposed to a link to your favourite online vendor :P. For systems like FATE, which are available online for free, such a link would be helpful. This is not that different from a link to boardgamegeek, which is how the field currently functions (though if you're connecting to the BGG API, storing the ID indeed makes more sense).

(I'm fine with leaving the code as-is)